src/Microsoft.CodeAnalysis.Analyzers/Core/MetaAnalyzers/DiagnosticDescriptorCreationAnalyzer.cs

```diff
@@ -305,18 +361,58 @@ bool IsCreateHelper(IMethodSymbol method)
                     method.Parameters[0].Type.SpecialType == SpecialType.System_String;
         }
 
-        private static void AnalyzeTitle(IMethodSymbol descriptorCreationMethod, IFieldInitializerOperation creation, Action<Diagnostic> reportDiagnostic)
+        private static void AnalyzeTitle(OperationAnalysisContext operationAnalysisContext, ImmutableArray<IArgumentOperation> creationArguments, IFieldInitializerOperation creation)
+        {
+            IArgumentOperation titleArgument = creationArguments.FirstOrDefault(a => a.Parameter.Name.Equals("title", StringComparison.OrdinalIgnoreCase));
+
+            if (titleArgument?.Parameter.Type.SpecialType == SpecialType.System_String)
+            {
+                operationAnalysisContext.ReportDiagnostic(creation.Value.CreateDiagnostic(UseLocalizableStringsInDescriptorRule, WellKnownTypeNames.MicrosoftCodeAnalysisLocalizableString));
+
+                if (titleArgument.Value.ConstantValue.HasValue &&
+                    titleArgument.Value.ConstantValue.Value is string title &&
+                    title.Length > 0 &&
+                    (title.Contains(". ") || title[title.Length - 1].Equals('.') || ContainsLineReturn(title)))
+                {
+                    operationAnalysisContext.ReportDiagnostic(titleArgument.CreateDiagnostic(DefineDiagnosticTitleCorrectlyRule));
+                }
+            }
+        }
+
+        private static void AnalyzeMessage(OperationAnalysisContext operationAnalysisContext, ImmutableArray<IArgumentOperation> creationArguments)
+        {
+            IArgumentOperation messageArgument = creationArguments.FirstOrDefault(a => a.Parameter.Name.Equals("messageFormat", StringComparison.OrdinalIgnoreCase));
+
+            if (messageArgument?.Value.ConstantValue.Value is string message &&
```

Comment:
Why no check for `messageArgument?.Value.ConstantValue.HasValue`? It might be better to have a simple static helper `bool TryGetNonEmptyConstantStringValue(IOperation operation, out string value)` that can be shared for all three?

Reply:
Ok I will create the helper. Regarding the check for `HasValue` I started with it but didn't really see the point as I was doing a safe-cast afterward anyways in case there is some non-compiling code.

Reply:
I think not doing a HasValue check will certainly cause Value access to throw if any of these parameters are passed non-constant string arguments.